slick-codegen/src/main/scala/slick/codegen/SourceCodeGenerator.scala

```diff
@@ -90,9 +91,11 @@ object SourceCodeGenerator {
       case uri :: outputDir :: Nil =>
         run(new URI(uri), Some(outputDir))
       case profile :: jdbcDriver :: url :: outputDir :: pkg :: Nil =>
-        run(profile, jdbcDriver, url, outputDir, pkg, None, None)
+        run(profile, jdbcDriver, url, outputDir, pkg, None, None, false)
       case profile :: jdbcDriver :: url :: outputDir :: pkg :: user :: password :: Nil =>
-        run(profile, jdbcDriver, url, outputDir, pkg, Some(user), Some(password))
+        run(profile, jdbcDriver, url, outputDir, pkg, Some(user), Some(password), false)
+      case  profile:: jdbcDriver :: url :: outputDir :: pkg :: user :: password :: ignoreInvalidDefaults :: Nil =>
+        run(profile, jdbcDriver, url, outputDir, pkg, Some(user), Some(password), Try(ignoreInvalidDefaults.toBoolean).getOrElse(false))
```

Comment:
not a fan of silently ignoring errors as in `Try(...).getOrElse(...)`. Let's let users know something went wrong. The easiest way would be simply throwing the exception, which I would be ok with.